### ISSUE TITLE
No applicative objects/queries shown

### ISSUE THREAD

**Issue by ptikf** [powa-team/powa-web#231]
In my current configuration (remote server, postgresql 16), 
- powa web ok, display all system statistics on distant databases,
- system queries on powa database displayes
- historical search works well

but: 
no applicative object displayed (messages: "no data found" on web)
no query on applicative object displayed (and no plans of course)
In query using system view, all paramaters replaced by bind variables ( $1, $2 and so). 

Where can I ibegin nvestigations ?

**Comment by ptikf** [comment]
Hello, this is a screenshot of my question :
- no data for database objects 
- only "technical" queries are shown 

![powa_q1](https://github.com/user-attachments/assets/98a8c5c9-da82-416b-9ae3-8a5fbe919fee)

**Comment by rjuju** [comment]
Thanks, I understand now!

> no applicative object displayed (messages: "no data found" on web)

The support for the "Database objects" datasource is not available for postgres 16 and above in powa 4.x.  It has been rewritten using a different approach and works with all versions with powa 5.0.  The other tabs should work though.

> no query on applicative object displayed (and no plans of course)

was there any activity on the chosen database for that remove server in the given interval?  I do see some user tables (timescaledb_information.hypertables", and there seems to be queries done by pgadmin4 (the pga4dash query at least), so there is definitely activity going on on that database.

> In query using system view, all paramaters replaced by bind variables ( $1, $2 and so).

That's pg_stat_statements behavior, which is the source of information for query activity in powa.  It normalises all the queries so that the query text doesn't have any actual parameter, and all statistics are aggregated on that normalized query.  You can refer to https://www.postgresql.org/docs/15/pgstatstatements.html for more details about it.

**Comment by ptikf** [comment]
Hello, screen capture for queries:
(db under test, some queries run, one table 4 billion rows):
![powa_q2](https://github.com/user-attachments/assets/8511bd09-a335-4bf0-8adb-a30f1ff40a4a)

- ok for new powa version  
- ok for pg_stat_statements
- but pg_qualstats should do the job ?

**Comment by rjuju** [comment]
pg_qualstats could be used to put some values on the parameters in the join and where clauses (but not in the select part of the statement), but it doesn't mean that we could associate specific metrics to those values. pg_stat_statements accumulates planning time, execution time and the rest of the counters to a single normalised entry, so the detail per specific parameter is never stored anywhere.

do you have a specific performance problem you're trying to solve?

**Comment by pgiraud** [comment]
If I'm not wrong, a query will not show in the table if it ran only once. Queries must have happened at least twice (and in different powa snapshot) to be taken into account.

**Comment by rjuju** [comment]
indeed, this is documented at https://powa.readthedocs.io/en/latest/FAQ.html#some-queries-don-t-show-up-in-the-ui
